### DISCUSSION TITLE
E-Mails are not sending & manage.py in the wrong place

### DISCUSSION THREAD

**Discussion by fshubthunfischbaum** [mediacms-io/mediacms#547] [Q&A]
**Describe the issue**
MediaCMS is not sending *any* E-Mails.

**To Reproduce**
Steps to reproduce the issue:
1. Enter E-Mail Server details
2. Register New User, no E-Mail

**Expected behavior**
Expecting the User to receive the registration E-Mail

**Environment (please complete the following information):**
 - OS:  Ubuntu 20.04.5 LTS (GNU/Linux 5.4.0-125-generic x86_64)
 - Singleserver
 - Firefox

**Additional context**
Could not test with the manage.py shell. Unable to load Django interface.
That manage.py is also not in the right folder. I did not move it, but the troubleshooting guide is based on different folder paths.
My location of the manage.py file: /home/mediacms.io/mediacms.py

I'm not sure, but there is a SSL Bug with a Version of smtplib of Python. Is this the issue for my installation as well?
I sent out an E-Mail without SSL/TLS. That went successfully.

**Comment by fshubthunfischbaum** [comment]
Hey everyone,
this is still an issue for us and is thus preventing our video platform from launch. Any help would be appreciated.
Thanks :)

**Comment by mgogoulos** [comment]
Hi, what  is the account provider you are using (eg Gmail, Mailgun etc)
And what are the logs? Can you do a `tail -f logs/celery*` , then try to send an email (eg use the forgot password) and paste here the error?

**Comment by fshubthunfischbaum** [comment]
Hi again! Thanks for your reply.
We are using our own Mail Service, just plain smtp. 

Output from the celery log:
`Invalid HTTP_HOST header: '${ip}:${port}'. The domain name provided is not valid according to RFC 1034/1035.`
The mailserverhost entered is mail.fshub.io with a 587 as port. This is the config:

```
DEFAULT_FROM_EMAIL = 'tube@removed'
EMAIL_HOST_PASSWORD = 'supersecretpasswordhere'
EMAIL_HOST_USER = 'tube@removed'
EMAIL_USE_TLS = True
SERVER_EMAIL = 'tube@removed'
EMAIL_HOST = 'removed'
EMAIL_PORT = 587
```
We should be good, right?



Edit: removed server details for privacy reasons.

**Comment by mgogoulos** [comment]
Hi, I'm not sure how this error is related with the email settings - which look good btw.

Can you share the complete trace?

Also there are some instructions here https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md#15-debugging-email-issues  that will help find the real problem, can you try that?

**Reply by fshubthunfischbaum** [reply]
uwe@fvidspace:/$ source /home/mediacms.io/bin/activate
(mediacms.io) uwe@fvidspace:/$ python manage.py shell
python: can't open file 'manage.py': [Errno 2] No such file or directory
(mediacms.io) uwe@vidspace:/$

**Reply by fshubthunfischbaum** [reply]
My manage.py is located /home/mediacms.io/mediacms$
is that correct?

**Reply by mgogoulos** [reply]
correct
```
uwe@fvidspace:/$ source /home/mediacms.io/bin/activate
uwe@fvidspace:/$ cd  mediacms
uwe@fvidspace:/mediacms$ python manage.py shell


```

**Reply by fshubthunfischbaum** [reply]
It returns 1.
This should mean that everything went fine?

**Reply by rbpawdev** [reply]
I am running into the same issue as OP.  My setup is exactly the same.

SMTP Server details in configuration file are fine, I have triple checked them.  I ran the test using the debugging instructions, and the email was sent successfully, which I take it to mean the credentials are good in the settings file, my network allows outbound messages and the smtp server is properly sending the message.  

However, when trying to register a user, an email is not sent.  Here is the error I see in the log: Cannot reach celery_email_backend

Any help is appreciated.

[ **EDIT: SOLVED** - A server reboot of the OS helped to resolve issue.  I guess the systemctl restart command was not covering all services that needed to be restarted after the configuration changes]

**Reply by fshubthunfischbaum** [reply]
The restart helped for me as well. Thank you @rbpawdev !
